src/Microsoft.ML.Dnn/TensorTypeExtensions.cs

```diff
@@ -3,49 +3,477 @@
 // See the LICENSE file in the project root for more information.
 
 using System;
+using System.Globalization;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Threading.Tasks;
 using Microsoft.ML.Internal.Utilities;
 using NumSharp.Backends;
-using NumSharp.Backends.Unmanaged;
 using NumSharp.Utilities;
 using Tensorflow;
 
+#if _REGEN_GLOBAL
+%supported_numericals = ["Boolean","Byte","Int16","UInt16","Int32","UInt32","Int64","UInt64","Double","Single"]
+%supported_numericals_lowercase = ["bool","byte","short","ushort","int","uint","long","ulong","double","float"]
+%supported_numericals_TF_DataType = ["TF_BOOL","TF_UINT8","TF_INT16","TF_UINT16","TF_INT32","TF_UINT32","TF_INT64","TF_UINT64","TF_DOUBLE","TF_FLOAT"]
+%supported_numericals_TF_DataType_full = ["TF_DataType.TF_BOOL","TF_DataType.TF_UINT8","TF_DataType.TF_INT16","TF_DataType.TF_UINT16","TF_DataType.TF_INT32","TF_DataType.TF_UINT32","TF_DataType.TF_INT64","TF_DataType.TF_UINT64","TF_DataType.TF_DOUBLE","TF_DataType.TF_FLOAT"]
+#endif
+
 namespace Microsoft.ML.Transforms
 {
     [BestFriend]
     internal static class TensorTypeExtensions
     {
         public static void ToScalar<T>(this Tensor tensor, ref T dst) where T : unmanaged
         {
```

Comment:
Can this handle string tensors?

Reply:
No because string is not unmanaged so it can't be used in a `T : unmanaged` constraint. This is why I provided a string-speicifc overloads which were requested for removal (and removed at https://github.com/dotnet/machinelearning/pull/4290/commits/effdc8aefb44862167e72e00996cf9d1307f4dfc) by @eerhardt

Reply:
I was sure I have conversion of `String-Tensor -> T`, Adding that one.